### ISSUE TITLE
[store] Expose `StoreCollection` as public

### ISSUE THREAD

**Issue by seokju-na** [tauri-apps/plugins-workspace#223]
Hello 👋

## Problems
Currently `StoreCollection` is not public, so cannot be access with app state.

There is a use case to directly control the store, and it would be nice to be able to access the app state.

```rust
fn setup<R: Runtime>(app: &mut App<R>)
{
  let stores = app.state::<State<'_, StoreCollection>>(); // access app state (expected)
}

fn main() {
  tauri::Builder::default()
    .plugin(PluginBuilder::default().build())
    .setup(setup)
    .run(tauri::generate_context!())
    .expect("error while running tauri application");
}
```

## Proposal

Expose `StoreCollection` as public:

```rust
#[derive(Default)]
pub struct StoreCollection {
  stores: Mutex<HashMap<PathBuf, Store>>,
  frozen: bool,
}
```

**Comment by JonasKruckenberg** [comment]
We have discussed and considered this many times before, but just exposing the StoreCollection as private has a number of issues:

1. No events are emitted. When you manipulate the Stores directly, these updates will not be send to the Frontend. (This doesn't break regular `get`/`set` just the `onChange` family of hooks)
2. You could sidestep the `frozen` mechanism of protecting against store creation. One could argue that's maybe a positive though?

So I'm a bit hesitant to just mark this as public as-is, but if we can find a better API than I'd be more than open to allowing this.

**Comment by seokju-na** [comment]
Thanks for your comments.
I understand mark as pubic is not a good way to go, completely agree with your comments.
I will think about whether there is a better API in other issue/PR. (closes tauri-apps/plugins-workspace#223)
